### DISCUSSION TITLE
Tuned usage

### DISCUSSION THREAD

**Discussion by younes-zeboudj** [nodejs/undici#2363] [General]
I use axios and fetch regularly and I want to switch to undici for a use case that requires more performance. How ever, I don't find a clear way through the doc to pass headers to the request or to catch the response headers before getting the body. Any thoughts?

**Comment by metcoder95** [comment]
I recommend you go through the [`Client`](https://undici.nodejs.org/#/docs/api/Client?id=new-clienturl-options) documentation to understand how to customize your request. If you want to intercept the headers before receiving a body or more complex operations, look at the `Client#dispatch` method, as it will allow you to control the lifecycle of your requests granularly.
